src/node_api.cc

```diff
@@ -741,8 +741,19 @@ napi_status napi_define_class(napi_env env,
     v8::Local<v8::String> property_name;
     CHECK_NEW_FROM_UTF8(isolate, property_name, p->utf8name);
 
+    // convert the properties from NAPI to v8 format
+    unsigned int attribute_flags = v8::None;
+    if ((p->attributes & napi_read_only)) {
+      attribute_flags |= v8::ReadOnly;
+    }
+    if ((p->attributes & napi_dont_enum)) {
+      attribute_flags |= v8::DontEnum;
+    }
+    if ((p->attributes & napi_dont_delete)) {
+      attribute_flags |= v8::DontDelete;
+    }
     v8::PropertyAttribute attributes =
-        static_cast<v8::PropertyAttribute>(p->attributes);
+        static_cast<v8::PropertyAttribute>(attribute_flags);
```

Comment:
There is a similar cast in `napi_define_properties()`, that should probably be changed too. (Generally: How about turning this into a general helper function that also takes care of the typecasting?)